java/code/src/com/redhat/rhn/manager/user/UserManager.java

```diff
@@ -148,6 +151,62 @@ public static boolean verifyPackagesAccess(Org org, List<Long> packageIds) {
         return (!dr.isEmpty());
     }
 
+    /**
+     * Get a concatenated stream of namespaces granted direct (to the user)
+     * or indirect (via access groups) access.
+     * @param user the user
+     * @return the stream of permitted namespaces
+     */
+    public static Stream<Namespace> getPermittedNamespaces(User user) {
+        return Stream.concat(user.getNamespaces().stream(),
+                user.getAccessGroups().stream().flatMap(g -> g.getNamespaces().stream()));
+    }
+
+    /**
+     * Ensures that the current user is permitted access to a namespace with the specified access mode
+     * @param user the user
+     * @param namespace the namespace
+     * @param mode the access mode (view/modify)
+     * @throws PermissionException if the user is not permitted to access the specified namespace
+     */
+    public static void ensureRoleBasedAccess(User user, String namespace, Namespace.AccessMode mode) {
```

Comment:
I think this method is not used. It also feels strange, since it doesn't return anything and checks user access and through an exception if he doesn't have access to the endpoint.

Reply:
Oh this one has a long story:

Imagine a struts page that renders a JSP table with pagination buttons, select all button, etc.
On top of that, there's a button to do some real action on selected rows.

In Struts, all the pagination and the actual action are handled in the same handler with the same POST endpoint. However, pagination is a read operation, while the action is usually a "modify".

Since the endpoint is the same, the only sensible way is to make the differentiation in the handler code.
So this method is called from these handlers. And the way it doesn't return but throws is for convenience.

Here's some example usages: https://github.com/uyuni-project/uyuni/commit/d4517b322fae994cc573dde90a28f4fc47cdc322